File: leetcode_python/235. Lowest Common Ancestor of a Binary Search Tree.py
class Solution:
    def lowestCommonAncestor(self, root: 'TreeNode', p: 'TreeNode', q: 'TreeNode') -> 'TreeNode':
        if p.val > q.val: p, q = q, p
        while root:
            if root.val < p.val: root = root.right
            elif root.val > q.val: root = root.left
            else: return root

File: leetcode_java/235. Lowest Common Ancestor of a Binary Search Tree.java
class Solution {
    public TreeNode lowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q) {
        int pv = Math.min(p.val, q.val), qv = Math.max(p.val, q.val);
        while (root != null) {
            if (root.val < pv) root = root.right;
            else if (root.val > qv) root = root.left;
            else return root;
        }
        return null;
    }
}